### ISSUE TITLE
[RFC]: add `@stdlib/array/base/take-map`

### ISSUE THREAD

**Issue by kgryte** [stdlib-js/stdlib#1321]
### Description

This RFC proposes adding the package `@stdlib/array/base/take-map`.

The package should be structured similarly to [`@stdlib/array/base/take`](https://github.com/stdlib-js/stdlib/tree/64931c5fdb4abc3a339b619658d3b5acc089e556/lib/node_modules/%40stdlib/array/base/take).

The package should expose an API having the same signature as `take`, with the addition of `clbk` and `thisArg` arguments.

```javascript
function takeMap( x, indices, mode, clbk[, thisArg] ) {...}
```

The callback function should map each selected value specified by `indices` to the output value. 

In short, this package combines `take` and `map` into a single API call.

Package: `@stdlib/array/base/take-map`
Alias: `takeMap`

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

**Comment by vr-varad** [comment]
https://github.com/stdlib-js/stdlib/pull/1349 solved
